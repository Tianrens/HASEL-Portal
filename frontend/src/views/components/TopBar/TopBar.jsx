import React from 'react';
import { Link } from 'react-router-dom';
import styles from './TopBar.module.scss';
import UoaLogo from '../../../assets/images/uoaLogo.svg';
import { HamburgerDrawer, NavLinks } from './NavLinks/TopBarHelpers';

export default function TopBar() {
    // TODO: Get userType and determine if they are admin or not
    const userType = 'SUPERADMIN';
    const isAdmin = true;

    return (
        <div className={styles.fillTopbarSpace}>
            <div className={styles.navbar}>
                <div className={styles.patternBgLayer}>
                    <div className={styles.bgLayer}>
                        <Link to='/'>
                            <img
                                className={styles.uoaLogo}
                                src={UoaLogo}
                                alt='University of Auckland'
                            />
                            <div className={styles.spacer} />
                            <div className={styles.title}>
                                HASEL Lab
                                {isAdmin ? <div className={styles.subTitle}>{userType}</div> : null}
                            </div>
                        </Link>

                        <div className={styles.pushSpacer} />

                        <div className={styles.horizontalNavLinks}>
                            <NavLinks isAdmin={isAdmin} />
                        </div>

                        <HamburgerDrawer isAdmin={isAdmin} />
                    </div>
                </div>
            </div>
        </div>
    );
}
