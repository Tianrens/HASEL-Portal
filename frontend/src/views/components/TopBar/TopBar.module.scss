@import '../../../assets/colours.module';
@import '../../../assets/responsiveDesignHelpers';

$topBarHeight: 58px;
$dropShadow: 0px 3px 5px rgba(0, 0, 0, 0.6);

.fillTopbarSpace {
    min-height: $topBarHeight;
    max-height: $topBarHeight;
}

.navbar {
    position: fixed;
    left: 0;
    right: 0;
    min-height: $topBarHeight;
    max-height: $topBarHeight;
    display: flex;
    flex-direction: row;
    align-items: center;
    color: white;
    z-index: 100;
    box-shadow: $dropShadow;
}

.patternBgLayer {
    @extend .navbar;
    background-color: $blackPatternBgColor;
    background-image: $blackGradientPatternBg;
}

.bgLayer {
    @extend .navbar;
    background: $blackGradient;
}

.uoaLogo {
    height: 65px;
    width: 65px;
    z-index: 2;
    position: absolute;
    left: 10px;
    top: 5px;
    filter: drop-shadow($dropShadow);
}

.spacer {
    width: 100px;
}

.title {
    color: white;
    text-shadow: 0px 2px 2px #262626;
    font-size: 1.8em;
    font-weight: bold;
    padding: 0.2em;
    flex-direction: column;
}

.subTitle {
    font-size: 0.3em;
    font-weight: normal;
    display: flex;
    justify-content: left;
}

.pushSpacer {
    flex: 1;
    justify-content: flex-end;
}

.horizontalNavLinks {
    display: block;
}

// Hide horizontal nav links in smaller views
@include responsive(tablet) {
    .horizontalNavLinks {
        display: none;
    }
}

@include responsive(tinyMobile) {
    .title {
        font-size: 1.2rem;
    }
    .subTitle {
        font-size: 0.5em;
    }
}
