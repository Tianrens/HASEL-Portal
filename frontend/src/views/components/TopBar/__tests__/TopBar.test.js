import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { StylesProvider } from '@material-ui/core';
import { MemoryRouter } from 'react-router-dom';
import TopBar from '../TopBar';

test('TopBar renders properly', () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshotComponent = shallowRenderer.render(
        <StylesProvider injectFirst>
            <MemoryRouter>
                <TopBar />
            </MemoryRouter>
        </StylesProvider>,
    );
    expect(snapshotComponent).toMatchSnapshot();
});
