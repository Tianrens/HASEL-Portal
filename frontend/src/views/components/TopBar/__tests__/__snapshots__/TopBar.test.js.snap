// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar renders properly 1`] = `
<Context.Provider
  value={
    Object {
      "disableGeneration": false,
      "generateClassName": [Function],
      "jss": Jss {
        "generateId": [Function],
        "id": 2,
        "options": Object {
          "Renderer": [Function],
          "createGenerateId": [Function],
          "id": Object {
            "minify": false,
          },
          "insertionPoint": <!--mui-inject-first-->,
          "plugins": Array [],
        },
        "plugins": PluginsRegistry {
          "plugins": Object {
            "external": Array [
              Object {
                "onCreateRule": [Function],
                "onProcessStyle": [Function],
                "onUpdate": [Function],
              },
              Object {
                "onCreateRule": [Function],
                "onProcessRule": [Function],
              },
              Object {
                "onProcessStyle": [Function],
              },
              Object {
                "onChangeValue": [Function],
                "onProcessStyle": [Function],
              },
              Object {
                "onChangeValue": [Function],
                "onProcessStyle": [Function],
              },
              Object {
                "onChangeValue": [Function],
                "onProcessRule": [Function],
                "onProcessStyle": [Function],
              },
              Object {
                "onProcessStyle": [Function],
              },
            ],
            "internal": Array [
              Object {
                "onCreateRule": [Function],
              },
              Object {
                "onCreateRule": [Function],
              },
              Object {
                "onChangeValue": [Function],
                "onCreateRule": [Function],
                "onProcessStyle": [Function],
              },
              Object {
                "onCreateRule": [Function],
              },
              Object {
                "onCreateRule": [Function],
              },
              Object {
                "onCreateRule": [Function],
              },
              Object {
                "onCreateRule": [Function],
              },
            ],
          },
          "registry": Object {
            "onChangeValue": Array [
              [Function],
              [Function],
              [Function],
              [Function],
            ],
            "onCreateRule": Array [
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
            ],
            "onProcessRule": Array [
              [Function],
              [Function],
            ],
            "onProcessSheet": Array [],
            "onProcessStyle": Array [
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
            ],
            "onUpdate": Array [
              [Function],
            ],
          },
        },
        "version": "10.7.1",
      },
      "sheetsCache": null,
      "sheetsManager": Map {},
      "sheetsRegistry": null,
    }
  }
>
  <MemoryRouter>
    <TopBar />
  </MemoryRouter>
</Context.Provider>
`;
